### PR TITLE
Improve static analysis

### PR DESCRIPTION
**Pull request by ammarfaizi2**
Improve static analysis

  - Make editorconfig-checker checks header files as well (not only C).
  - Implement better indentation checker tool.
  - Clean up missed code styles reported by these shiny tools.

Fixes: https://github.com/GNUWeeb/GNUWeebBot/issues/30
Signed-off-by: Ammar Faizi <ammarfaizi2@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -391,5 +391,12 @@ run_vg: $(TARGET_BIN)
 	$(VG) $(VGFLAGS) ./$(TARGET_BIN) -c config_1.ini
 
 
+tools/indent_check: tools/indent_check.c
+	$(CC) $(<) -Wall -Wextra -O3 -o $(@)
```
Comment (komori-k) [inline Makefile:395]: Can't we use `CFLAGS` here?

File: src/gwbot/include/gwbot/lib/tg_event.h [hunk 2]
```diff
@@ -412,19 +408,19 @@ static __always_inline int parse_tgevi_chat(json_object *jchat,
...
 								&res))) {
 				pr_err("Cannot find \"all_admins\" "
 						"on key \"chat\"");
 				return -EINVAL;
 
 			}
-			chat->all_admins = 
+			chat->all_admins =
 				json_object_get_boolean(res)  ? true : false;
 		}
```
Comment (komori-k) [inline src/gwbot/include/gwbot/lib/tg_event.h:421]: Doesn't `json_object_get_boolean` return `bool`?
Do we need this ternary check?

File: src/gwbot/include/gwbot/lib/tg_event/gif.h [hunk 3]
```diff
@@ -118,24 +118,24 @@ static __always_inline int parse_event_gif(json_object *jgif,
 
 	if (egif->is_forwarded)
 	{
-		if (unlikely(!json_object_object_get_ex(jgif,
```
Comment (komori-k) [inline src/gwbot/include/gwbot/lib/tg_event/gif.h:121]: Wrong brace placement

File: tools/indent_check.c [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+	while ((bp = fgets(buffer, sizeof(buffer), handle))) {
+		lineno++;
+		if (bp[0] == ' ' && bp[1] != '*') {
+			ret = 1;
+			printf("  Bad indentation char detected: %s:%u\n", fn,
+			       lineno);
+		}
+	}
+	fclose(handle);
```
Comment (komori-k) [inline tools/indent_check.c:37]: I wonder if we can add 80 chars per line limit check

File: tools/indent_check.c [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+static int indent_check(const char *fn)
+{
+	int ret = 0;
+	uint32_t lineno = 0;
+	char buffer[8192], *bp;
+	FILE *handle = fopen(fn, "rb");
```
Comment (komori-k) [inline tools/indent_check.c:19]: `sizeof(buffer)` seems too big, don't you think so?

**Review by komori-k — CHANGES_REQUESTED** [review]

**Comment by ammarfaizi2** [comment]
Good catch @komori-k, I am addressing your review, will push the fix soon. Please wait for it.

**Comment by ammarfaizi2** [comment]
I will ignore the buffer size part right now. And then for 80 chars per line check, probably I can ship it in the next PR. So I will address it later.

**Comment by ammarfaizi2** [comment]
The part that compiles `tools/indent_check.c` is fine not to use `CFLAGS` variable. I want that file is always compiled with optimization enabled and simple flags. No need so many complicated flags.

**Comment by ammarfaizi2** [comment]
I will add `-Wpedantic` to improve the warning quality.

**Pull request merged**